src/main/scala/viper/gobra/frontend/info/implementation/typing/ExprTyping.scala

```diff
@@ -284,15 +297,16 @@ trait ExprTyping extends BaseTyping { this: TypeInfoImpl =>
             case t: AbstractType => t.messages(n, n.args map exprType)
             case t => error(n, s"type error: got $t but expected function type or AbstractType")
           }
-          pureReceiverMsgs ++ pureArgsMsgs ++ argAssignMsgs
+          error(n, "Cannot reveal a predicate call.", n.reveal) ++ pureReceiverMsgs ++ pureArgsMsgs ++ argAssignMsgs
 
         case (Left(callee), Some(_: ap.PredExprInstance)) =>
-          exprType(callee) match {
+          val wellTypedArguments = exprType(callee) match {
             case PredT(args) =>
               if (n.args.isEmpty && args.isEmpty) noMessages
               else multiAssignableTo.errors(n.args map exprType, args)(n) ++ n.args.flatMap(isExpr(_).out)
             case c => Violation.violation(s"This case should be unreachable, but got $c")
           }
+          error(n, "Cannot reveal a predicate expression instance.", n.reveal) ++ wellTypedArguments
```

Comment:
What is the difference between a predicate call and a predicate expression instance? Currently, we only have a test that results in the error "Cannot reveal a predicate call" (reveal-predicate-fail1.gobra), but not one for "Cannot reveal expression instance".

Reply:
Should we add a test for exercising "Cannot reveal expression instance"? If yes, what should it look like?